### ISSUE TITLE
Properties should be inferred to exist on object after "in" expression

### ISSUE THREAD

**Issue by schuelermine** [microsoft/TypeScript#43856]
# Suggestion

## 🔍 Search Terms

in any unknown property access exist infer

## ✅ Viability Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, new syntax sugar for JS, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

## ⭐ Suggestion

Accessing a property in a block where an `in` expression involving the property name is guaranteed to be true shouldn't give a typecheck error.

## 📃 Motivating Example

I have a function that checks any object for something.  
I'd like to avoid using `any` as the argument type, because it completely disables typechecking.
If I do this:
```ts
function foo(x: object): boolean {
    if ("m" in x) {
        return typeof x.m === "object"
    } else {
        return false
    }
}
```
this is viewed as invalid by TS.
With this new feature, it's valid, because `"m" in x` being true guarantees that `x.m` won't fail.

## 💻 Use Cases

Type guards for some types that don't use `any`.

**Comment by MartinJohns** [comment]
Duplicate of #21732.

**Comment by schuelermine** [comment]
@MartinJohns 
Thanks!
